end-to-end-tests/src/helpers/mod.rs

```diff
@@ -0,0 +1,37 @@
+pub mod ctx;
+
+use anyhow::{Context, Result};
+use oxide_client::types::Name;
+use rand::{thread_rng, Rng};
+use std::future::Future;
+use std::time::{Duration, Instant};
+
+pub fn generate_name(prefix: &str) -> Result<Name> {
+    format!("{}-{:x}", prefix, thread_rng().gen_range(0..0xfff_ffff_ffffu64))
+        .try_into()
+        .map_err(anyhow::Error::msg)
+}
+
+pub async fn try_loop<F, Fut, T, E>(mut f: F, timeout: Duration) -> Result<T>
```

Comment:
You might find `test-utils::dev::poll::wait_for_condition` useful here?

Reply:
Oh, yep, this is basically what I wrote. I'll defer to the existing one :)